### PR TITLE
Improve API naming, README and Changelog

### PR DESCRIPTION
**Pull request by etagwerker**
Hey @bronzdoc, 

This PR improves the documentation and the method naming in the library. 

Also, it removes the ChangeLog.md details that were talking about `bundler-leak`

Please check it out. 

Thanks!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,13 +9,13 @@
 
 ## Description
 
-Patch-level verification for [bundler].
+The best tool to find leaky gems in your dependencies. Make sure your memory leaks
```
Comment (bronzdoc) [inline README.md]: Should this be "Make sure memory leaks"  instead of " Make sure your memory leaks" ?

Comment (etagwerker) [reply README.md]: @bronzdoc That sounds better. I just updated this PR.

File: README.md [hunk 2]
```diff
@@ -9,13 +9,13 @@
...
 
-Patch-level verification for [bundler].
+The best tool to find leaky gems in your dependencies. Make sure memory leaks
+are not in your gem dependencies.
 
 ## Features
 
-* Checks for memory leaks of gems in `Gemfile.lock`.
-* Prints memory leak information.
-* Does not require a network connection.
```
Comment (bronzdoc) [inline README.md]: Why this ` Does not require a network connection.` was removed?

Comment (etagwerker) [reply README.md]: @bronzdoc I don't think it's accurate, I guess it doesn't require network connection after doing `gem install bundler-leak` ... 

So, I don't know, is this line here to describe that when you run `bundler-leak` it won't connect to a remote server to fetch the ruby mem advisory db?

Comment (bronzdoc) [reply README.md]: > So, I don't know, is this line here to describe that when you run bundler-leak it won't connect to a remote server to fetch the ruby mem advisory db?

Yes, bundler-leak has it's own local copy

**Pull request merged**